src/core/src/core_logic/ServiceManager.py

```diff
@@ -0,0 +1,139 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+
+""" ServiceManager """
+import os
+from core.src.bootstrap.Constants import Constants
+
+
+class ServiceManager(object):
+    def __init__(self, env_layer, execution_config, composite_logger, telemetry_writer, service_info):
+        self.env_layer = env_layer
+        self.execution_config = execution_config
+        self.composite_logger = composite_logger
+        self.telemetry_writer = telemetry_writer
+
+        self.service_name = service_info.service_name
+        self.service_desc = service_info.service_desc
+        self.service_exec_path = service_info.service_exec_path
+
+        self.__systemd_path = "/run/systemd/system/"
+        self.__systemd_service_unit_path = "/etc/systemd/system/{0}.service"
+
+        self.service_start_cmd = "sudo systemctl start {0}.service"
+        self.service_stop_cmd = "sudo systemctl stop {0}.service"
+        self.service_reload_cmd = "sudo systemctl reload-or-restart {0}.service"
+        self.service_enable_cmd = "sudo systemctl enable {0}.service"
+        self.service_disable_cmd = "sudo systemctl disable {0}.service"
+        self.service_is_enabled_cmd = "sudo systemctl is-enabled {0}.service"
+        self.service_is_active_cmd = "sudo systemctl is-active {0}.service"
+
+        self.systemctl_daemon_reload_cmd = "sudo systemctl daemon-reload"
+        self.systemctl_version = "systemctl --version"
+
+    def remove_service(self):
+        service_path = self.__systemd_service_unit_path.format(self.service_name)
+        if os.path.exists(service_path):
+            self.stop_service()
+            self.disable_service()
+            os.remove(service_path)
+            self.systemctl_daemon_reload()
+
+    def create_and_set_service_idem(self):
+        """ Idempotent creation and setting of the service associated with the service the class is instantiated with """
+        self.remove_service()
+        self.create_service_unit_file(exec_start="/bin/bash " + self.service_exec_path, desc="Microsoft Linux Patch Extension - Auto-assessment")
+        self.systemctl_daemon_reload()
+        self.enable_service()
+        self.start_service()
+
+    # region - Service Management
+    def start_service(self):
+        code, out = self.__invoke_systemctl(self.service_start_cmd.format(self.service_name), "Starting the service.")
+        return code == 0
+
+    def stop_service(self):
+        code, out = self.__invoke_systemctl(self.service_stop_cmd.format(self.service_name), "Stopping the service.")
+        return code == 0
+
+    def reload_service(self):
+        code, out = self.__invoke_systemctl(self.service_reload_cmd.format(self.service_name), "Reloading the service.")
+        return code == 0
+
+    def get_service_status(self):
```

Comment:
Is command here is same as stop_service? both stop_service and get_service_status same command is used

Reply:
Adding later as not needed for mainline scenario..